pkg/engine/printers/generate.go

```diff
@@ -15,3 +16,12 @@ func init() {
 func Generate(obj runtime.Object) (string, bool) {
 	return tg.GenerateTable(obj)
 }
+
+// PrintCustomizedHealthCheck prints customized health check result defined in the `extensions` field of the resource in the Spec.
+func PrintCustomizedHealthCheck(healthPolicyCode string, resource []byte) (string, bool) {
+	err := kcl.RunKCLHealthCheck(healthPolicyCode, resource)
+	if err != nil {
+		return "Reconciling...", false
```

Comment:
> Here it may be necessary to judge the returned error, distinguishing whether it is due to the failure to meet the health check policy or to the error in the `KCL` grammar itself. For the latter condition, the watch process may need to be terminated and the error should be thrown upwards. Or the validation of the health check policy grammar can be placed before.


For the latter condition mentioned above, the `watch` process will be skipped so that it does not stop the `apply` process. In all other failure cases, the error details will be printed in the table during the `apply` process until either the `apply` is successful or the `apply` reaches a timeout condition. This is done this way because some of the health check validations can only succeed after successful resource deployment, though the KCL assertions could be incorrect in the first place. This implementation currently has limitations but we will strive for a better solution in the future.